phalcon/db/adapter/pdo.zep

```diff
@@ -85,9 +85,12 @@ abstract class Pdo extends Adapter
 	 */
 	public function connect(descriptor = null)
 	{
-		var username, password, dsnParts, dsnAttributes,
+		var eventsManager,
+			username, password, dsnParts, dsnAttributes,
 			persistent, options, key, value;
 
+		let eventsManager = <ManagerInterface> this->_eventsManager;
```

Comment:
What if `setEventsManager` was removed from the adapter and it had to be passed into the constructor.  Its a small breakage but then all events would always work.  Also people would be able to instantly see the error message about the missing method and then to read the upgrade guide.  That kind of breakage on upgrade is very easy to fix.